src/chains/optimism/vm/opcodes/block/number.ts

```diff
@@ -0,0 +1,28 @@
+import { Opcode } from '@/chains/types';
+
+export const number: Opcode = {
+  number: 43,
+  name: 'number',
+  description: "Get the L2 block's number",
```

Comment:
Hmm, should `NUMBER` and `TIMESTAMP` be considered different for optimism? You can argue the opcode is identical on both chains (mainnet and optimism) because it returns the block number/timestamp of it's own chain, whereas if it returned the L1 block number (like arbitrum) that'd be considered different.

Counter to this, the [Optimism docs](https://community.optimism.io/docs/developers/build/differences/) do list `NUMBER` and `TIMESTAMP` as different. But from a developer's perspective, the opcodes behave the same as they do on mainnet, so I lean towards considering them the same (i.e. no diff)

Reply:
Yes, I realise that this was not the best opcode to show that there was a diff. I also think they should be considered the same for both Optimism and Arbitrum.